applications/luci-app-squid/htdocs/luci-static/resources/view/squid.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+'require form';
+'require fs';
+'require rpc';
+'require uci';
+'require view';
+
+var getOptions = rpc.declare({
+	object: 'luci.squid',
+	method: 'getOptions',
+	expect: { options: [] }
+});
+
+function validateFile(path) {
+	if (!path.startsWith('/etc/squid/')) {
+		return _('File must be located in directory /etc/squid');
+	}
+	return true;
+}
+
+function writeFile(path, content) {
+	if (content) {
+		var normalized = content.replaceAll('\r\n', '\n');
+		fs.write(path, normalized);
+	}
+}
+
+return view.extend({
+
+	load: function() {
+		var load_squid = uci.load('squid')
+			.then(() => uci.get('squid', 'squid'));
+		return Promise.all([load_squid, getOptions()]);
+	},
+
+	render: function(data) {
+		var { config_file, mime_table } = data[0];
+		var options = data[1];
+
+		var m, s, o;
+
+		m = new form.Map('squid', _('Squid'));
+
+		s = m.section(form.TypedSection, 'squid');
+		s.anonymous = true;
+		s.addremove = false;
+
+		s.tab('general', _('General Settings'));
+		s.tab('advanced', _('Advanced Settings'));
+
+		var o = s.taboption('general', form.Value, 'config_file', _('Config file'));
+		o.datatype = 'string';
+		o.default = '/etc/squid/squid.conf';
+		o.validate = function(section_id, value) {
+			return validateFile(value);
+		};
+
+		o = s.taboption('general', form.Value, 'mime_table', _('Mime table'));
+		o.datatype = 'string';
+		o.default = '/etc/squid/mime.conf';
+		o.validate = function(section_id, value) {
+			return validateFile(value);
+		};
+
+		o = s.taboption('general', form.Value, 'http_port', _('Port'));
+		o.datatype = 'portrange';
+		o.placeholder = '0-65535';
+
+		o = s.taboption('general', form.Value, 'http_port_options', _('HTTP port options'));
+		o.datatype = 'string';
+		o.optional = true;
+
+		o = s.taboption('general', form.Value, 'ssl_db', _('SSL DB'));
+		o.datatype = 'string';
+		o.optional = true;
+
+		o = s.taboption('general', form.Value, 'ssldb_options', _('SSL DB options'));
+		o.datatype = 'string';
+		o.optional = true;
+
+		o = s.taboption('general', form.Value, 'visible_hostname', _('Visible Hostname'));
+		o.datatype = 'string';
+		o.placeholder = 'OpenWrt';
+
+		o = s.taboption('general', form.Value, 'coredump_dir', _('Coredump files directory'));
+		o.datatype = 'string';
+		o.placeholder = '/tmp/squid';
+
+		var enable_icmp_option = '--enable-icmp';
```

Comment:
```js
		var enable_icmp_option = '<code>--enable-icmp</code>';
```

Reply:
That variable is for mapping the option in the rpcd output.

```
root@labmouse:~# ubus call luci.squid getOptions
{
        "options": [
                "--target=x86_64-openwrt-linux",
                "--host=x86_64-openwrt-linux",
                "--build=x86_64-pc-linux-gnu",
                "--disable-dependency-tracking",
                "--program-prefix=",
                "--program-suffix=",
                "--prefix=/usr",
                "--exec-prefix=/usr",
                "--bindir=/usr/bin",
                "--sbindir=/usr/sbin",
                "--libexecdir=/usr/lib",
                "--sysconfdir=/etc",
                "--datadir=/usr/share",
               "--enable-icmp", # Would show up here if set 
                "<many, many more options here>"
        ]
}
```

I assume you meant something like this?

```diff
diff --git a/applications/luci-app-squid/htdocs/luci-static/resources/view/squid.js b/applications/luci-app-squid/htdocs/luci-static/resources/view/squid.js
index abd2472e72..93d443d0f2 100644
--- a/applications/luci-app-squid/htdocs/luci-static/resources/view/squid.js
+++ b/applications/luci-app-squid/htdocs/luci-static/resources/view/squid.js
@@ -90,7 +90,7 @@ return view.extend({
                var enable_icmp_option = '--enable-icmp';
                var is_enable_icmp_defined = options.includes(enable_icmp_option);
                o = s.taboption('general', form.Flag, 'pinger_enable', _('Enable ICMP pinger'),
-                       !is_enable_icmp_defined ? _('Can only be set if Squid is compiled with the %s option').format(enable_icmp_option) : null);
+                       !is_enable_icmp_defined ? _('Can only be set if Squid is compiled with the <code>--%s</code> option').format(enable_icmp_option) : null);
                o.datatype = 'string';
                o.enabled = 'on';
                o.disabled = 'off';

Reply:
Ah, yes, right. But we prefer to avoid overloading i18n peeps with formatting gunk. 

Perhaps this?:
```js
			!is_enable_icmp_defined ? _('Can only be set if Squid is compiled with the %s option'.format('<code>%s</code>').format(enable_icmp_option)) : null);
```

Reply:
In any case, your way of scraping the CT options is really nice. I'm probably going to steal this for use elsewhere 😄